lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -787,6 +787,39 @@ class JoinConversionPattern : public HandshakeConversionPattern<JoinOp> {
   };
 };
 
+class SyncConversionPattern : public HandshakeConversionPattern<SyncOp> {
+public:
+  using HandshakeConversionPattern<SyncOp>::HandshakeConversionPattern;
+  void buildModule(SyncOp op, BackedgeBuilder &bb, RTLBuilder &s,
+                   hw::HWModulePortAccessor &ports) const override {
+    auto unwrappedIO = unwrapIO(s, bb, ports);
+
+    // Create something like a wire which can be feed to the builders
+    auto i1Type = s.b.getI1Type();
+    auto valid = std::make_shared<Backedge>(bb.get(i1Type));
+    auto ready = std::make_shared<Backedge>(bb.get(i1Type));
+    auto data = std::make_shared<Backedge>(bb.get(s.b.getNoneType()));
```

Comment:
I'd be in favor of factoring this out. Seems like a general use-case (creating a dataflow "wire").

Reply:
One just has to be careful when creating the corresponding in- and outputs, as one otherwise works with outdated values.

Reply:
Does the added struct look reasonable?